gcp-common/src/test/groovy/io/micronaut/gcp/credentials/GoogleCredentialsFactorySpec.groovy

```diff
@@ -1,16 +1,65 @@
 package io.micronaut.gcp.credentials
 
+import com.google.auth.RequestMetadataCallback
 import com.google.auth.oauth2.GoogleCredentials
+import com.google.auth.oauth2.ImpersonatedCredentials
+import com.google.auth.oauth2.ServiceAccountCredentials
+import com.google.auth.oauth2.UserCredentials
+import com.google.common.util.concurrent.MoreExecutors
 import io.micronaut.context.ApplicationContext
+import io.micronaut.context.annotation.Requires
+import io.micronaut.context.exceptions.BeanInstantiationException
+import io.micronaut.context.exceptions.ConfigurationException
 import io.micronaut.context.exceptions.NoSuchBeanException
+import io.micronaut.core.reflect.ReflectionUtils
```

Comment:
We have to be careful. we are coping GCP tests to an internal class.

Reply:
Understood. This was absolutely used as a last resort. It is a workaround for the fact that `GoogleCredentials` is internally caching the credentials loaded from a given .json file in a static instance of `DefaultCredentialsProvider` the first time the .json is loaded. Using this hack in `cleanup()` is unfortunately the only good way to make these test methods free of side-effects and repeatable.

This tight coupling to the internal details of `GoogleCredentials` already exists here and is one of the things that motivated me to add these tests in the first place. Better to have `GoogleCredentials` change and break these tests than to have that change in behavior make the module stop working correctly in a deployed app. :)